### ISSUE TITLE
Web storage missing

### ISSUE THREAD

**Issue by mrclauss** [isar-community/isar#50]
### Steps to Reproduce

Currently web in v4 only supports in-memory storage, no persistence.
Needs to be fixed

### Code sample

```dart
Provide a few simple lines of code to show your problem.
```

### Details

 - Platform: iPhone 13 Pro, Galaxy S7, x86 Android Emulator on Windows etc.
 - Flutter version: [e.g. 3.0.0]
 - Isar version: [e.g. 2.5.0]

---

 - [ ] I searched for similar issues already
 - [ ] I filled the details section with the exact device model and version
 - [ ] I am able to provide a reproducible example

**Comment by mdmm13** [comment]
Discussion with options: https://t.me/isardb/6339/8816: 

> Oh, one thing I could really use help: Isar currently only supports in memory SQLite on web. To persist data we need to write a custom [virtual](https://www.sqlite.org/vfs.html) [file](https://www.sqlite.org/vfs.html) [system](https://www.sqlite.org/vfs.html) [VFS](https://www.sqlite.org/vfs.html) that supports persistence.
> 
> We can either use IndexedDB to emulate a file system or we use the new [Origin Private File system](https://web.dev/origin-private-file-system/). There is some projects we can base our work on:
> 
> - Drift has [multiple VFSes](https://github.com/simolus3/sqlite3.dart/tree/a8d870c971d50e9023b88ad9ea7ae466af74b502/sqlite3/lib/src/wasm/vfs). I think none of them is perfect but we can learn from them. [Here are docs](https://drift.simonbinder.eu/web/) about them.
> - [wa-sqlite](https://github.com/rhashimoto/wa-sqlite/tree/master/src/examples) is a collection of javascript VFSes. Check out the discussions in the repository to learn more about them.
> - And of course [absurd sql](https://github.com/jlongster/absurd-sql) and the [corresponding blog post](https://web.archive.org/web/20230528161353/https://jlongster.com/future-sql-web). It's the most advanced one of them.
> 
> If you have experience with SQLite and javascript and would like to start a discussion/work on this let me know 💜

**Comment by mrclauss** [comment]
Yes I remember the discussion. In my opinion the most promising way to tackle this would be to adapt how drift/sqlite3 realised it. Drift web is announced as stanle, wo the same mechabism should work for isar.

Am 13. April 2024 13:24:34 MESZ schrieb mdmm13 ***@***.***>:
>Discussion with options: https://t.me/isardb/6339/8816: 
>
>> Oh, one thing I could really use help: Isar currently only supports
>in memory SQLite on web. To persist data we need to write a custom
>[virtual](https://www.sqlite.org/vfs.html)
>[file](https://www.sqlite.org/vfs.html)
>[system](https://www.sqlite.org/vfs.html)
>[VFS](https://www.sqlite.org/vfs.html) that supports persistence.
>> 
>> We can either use IndexedDB to emulate a file system or we use the
>new [Origin Private File
>system](https://web.dev/origin-private-file-system/). There is some
>projects we can base our work on:
>> 
>> - Drift has [multiple
>VFSes](https://github.com/simolus3/sqlite3.dart/tree/a8d870c971d50e9023b88ad9ea7ae466af74b502/sqlite3/lib/src/wasm/vfs).
>I think none of them is perfect but we can learn from them. [Here are
>docs](https://drift.simonbinder.eu/web/) about them.
>> -
>[wa-sqlite](https://github.com/rhashimoto/wa-sqlite/tree/master/src/examples)
>is a collection of javascript VFSes. Check out the discussions in the
>repository to learn more about them.
>> - And of course [absurd sql](https://github.com/jlongster/absurd-sql)
>and the [corresponding blog
>post](https://web.archive.org/web/20230528161353/https://jlongster.com/future-sql-web).
>It's the most advanced one of them.
>> 
>> If you have experience with SQLite and javascript and would like to
>start a discussion/work on this let me know 💜
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/isar-community/isar/issues/50#issuecomment-2053618439
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>

-- 
Martin Clauss
Meine Contactulater Card - All meine Kontaktinfo immer aktuell und vollständig:
https://contactulater.app/me/mc

**Comment by rackberg** [comment]
Any progress here?
We also need isar v4 persistence storage in web.

Why is it so complicated to implement the feature like drift/sqlite3?
Can someone explain it to me? I don't get it (no 'im not an expert with virtual filesystems using flutter). 

Has someone tried to reach out to @simolus3 (aka Simon Binder, developer of drift) and asked him for help?

Thanks in advance.
